lib/honeybadger.ex

```diff
@@ -161,7 +170,26 @@ defmodule Honeybadger do
       hostname: :inet.gethostname |> elem(1) |> List.to_string,
       origin: "https://api.honeybadger.io",
       project_root: System.cwd,
-      use_logger: true,
-      environment_name: nil]
+      use_logger: true]
+  end
+
+  defp get_and_set_environment_name!(config) do
+    case System.get_env("MIX_ENV") do
```

Comment:
I went back and forth when I was working on this. Ultimately I ended up going with the environment variable having preference because that is more consistent with other tools that provide overridable configuration via env vars. I can see it either way though. Do the other Honeybadger clients have existing precedence with this?

> On Feb 17, 2016, at 1:30 PM, joshuap <notifications@github.com> wrote:
> 
> In lib/honeybadger.ex:
> 
> > @@ -161,7 +170,26 @@ defmodule Honeybadger do
> >        hostname: :inet.gethostname |> elem(1) |> List.to_string,
> >        origin: "https://api.honeybadger.io",
> >        project_root: System.cwd,
> > -      use_logger: true,
> > -      environment_name: nil]
> > +      use_logger: true]
> > +  end
> > +
> > +  defp get_and_set_environment_name!(config) do
> > +    case System.get_env("MIX_ENV") do
> I'm thinking that maybe MIX_ENV should be the default, but if the honeybadger config option is set explicitly, that should be preferred. It looks like this will always prefer the environment variable. @rbishop thoughts?
> 
> —
> Reply to this email directly or view it on GitHub.
>